### PR TITLE
Add Battery Widget

### PR DESCRIPTION
**Pull request by mosswg**
Added a battery widget which only shows if the battery folder is set in the config and exists.

### CODE REVIEW

File: src/System.cpp [hunk 1]
```diff
@@ -141,6 +146,23 @@ namespace System
         return temp;
     }
 
+    double GetBatteryPercentage()
+    {
+        std::ifstream fullChargeFile(config.batteryFolder + "/charge_full");
+        std::ifstream currentChargeFile(config.batteryFolder + "/charge_now");
+        if (!fullChargeFile.is_open() || !currentChargeFile.is_open())
+        {
+            return 0.f;
```
Comment (scorpion-26) [inline src/System.cpp:155]: Can you make the result negative on error? Since *technically* (although very unlikely) 0 is a valid percentage.

**Review by scorpion-26 — COMMENTED** [review]
Thanks for the PR!
Can you add the new config to the example config(data/config)?

Other than that, looks good to me.

**Comment by mosswg** [comment]
Sure, I've done both the css and the config in my .config folder but I forgot to add to the repo. I can commit those in about an hour. I don't have access to my computer right now.

**Comment by mosswg** [comment]
Just added the CSS and config. I used pink for the CSS since it seemed to be the only unused color. Hope everything looks good, I've never used Sass before.

**Comment by scorpion-26** [comment]
Everything looks good!
I'll merge, once the source code comment I made is addressed.

**Comment by mosswg** [comment]
Sorry, I missed that. It returns -1 now.

**Comment by scorpion-26** [comment]
One small thing, can you check for > 0 instead of >= 1?

**Comment by scorpion-26** [comment]
Thanks!

**Pull request merged**